modules/openapi-generator/src/main/resources/go-experimental/model.mustache

```diff
@@ -38,6 +38,19 @@ type {{classname}} struct {
 	{{name}} {{^required}}*{{/required}}{{{dataType}}} `json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
 {{/vars}}
 }
+
```

Comment:
can you add a docstring?

Reply:
Any reason not to require the required Params in the input?

Reply:
@jirikuncar thanks for pointing out the docs I'll fix that.

@platinummonkey I think the problem here is that it's not straightforward what to do with required params that have defaults set. Should these be required arguments for the constructor or not? I think that this uncertainty is why generated constructors in other similar languages don't have required params either. I'd like to stay consistent here, so that's why I chose this approach.

Reply:
I pushed a new commit that adds both the docstring and also documentation in `model_doc`.

Reply:
`New(args....)` vs `NewDefault()` maybe is more clear?

Reply:
@platinummonkey I just realized one more reason that is problematic with the required properties:

Imagine that you have required properties A and B and code generated for that: `NewFoo(A, B)`. Now a change happens that makes `B` optional. You regenerate the code and suddenly you get `NewFoo(A)` - so that's backwards incompatible. But the change you did in API *is* backwards compatible, so this breaks the promise of not breaking API of the generated client when a non-breaking API change happens.

Reply:
I'm not actually sure that's a problem, if you need to introduce a var, this this method signature is also changed. The alternative is to list all variables, and if one becomes non-optional though or a new one is added then still the signature is changed. Seem moot given go's syntax.

Reply:
The signature of this method will never change, as it never accepts any arguments. But I think I'm starting to like the solution proposed in your previous comment with `New(required, args, ...)` and `NewWithDefaults()`. Assuming we clearly document that the contract of the former will always change when the set of required arguments changes, it should be ok. I'll work on this.